test/Microsoft.ML.Tests/ScenariosWithDirectInstantiation/TensorflowTests.cs

```diff
@@ -1334,6 +1381,12 @@ public void TensorFlowImageClassification(ImageClassificationTrainer.Architectur
                     .Fit(testDataset)
                     .Transform(testDataset);
 
+            // Check if the bottleneck cached values already exist
+            string trainSetBottleneckCachedValuesFileName = "TrainsetCached_" + finalImagesFolderName + "_" + (int)arch;
+            string validationSetBottleneckCachedValuesFileName = "validationsetCached_" + finalImagesFolderName + "_" + (int)arch;
+            string workspacePath = Path.Combine(TensorFlowScenariosTestsFixture.parentWorkspacePath, finalImagesFolderName + "_" + (int)arch);
+            bool isReuse = ShouldReuse(workspacePath, trainSetBottleneckCachedValuesFileName, validationSetBottleneckCachedValuesFileName);
```

Comment:
You can make a util function to common this out. Rule of thumb: Anytime 5 or more line are repeated make it in a function and reuse, not only it helps with maintainability of the code but also readability.